providers/base/units/eeprom/jobs.pxu

```diff
@@ -0,0 +1,39 @@
+id: eeprom/read-write
+_summary: Test EEPROM read and write functions for the system. 
+plugin: shell
+user: root
+category_id: eeprom 
+flags: also-after-suspend
+estimated_duration: 5
+imports: from com.canonical.plainbox import manifest
+requires: manifest.has_eeprom == 'True'
+command:
+  eeprom_path=$(find /sys/ -name eeprom)
+  if [ ! "$eeprom_path" ];
+  then
+    echo "Error: Can't find EEPROM in system!!"
+    exit 1
+  fi
+  status=0
+  for eeprom in $eeprom_path
+  do
+    data=$(echo $RANDOM | md5sum | head -c 10)
+    echo "Write data ${data} into ${eeprom}"
+    if (echo "$data" > "$eeprom")
+    then
+      echo "Write data ${data} successfully"
+      echo "Read data back from EEPROM"
+      read_data=$(cat < "$eeprom" | od -c | awk 'NR==1{for(i=2;i<=11;i++) printf $i}')
```

Comment:
I bet this was hard to write. Could you add some comment explaining what it does and why?

Reply:
Hi @kissiel,
The data read from EEPROM is like [this](https://pastebin.canonical.com/p/d3YfgRdVvw/).
And read with `od -c` will be like [this](https://pastebin.canonical.com/p/s8wjMYrNrh/), and this data becomes readable by humans.
Therefore, this command `read_data=$(cat < "$eeprom" | od -c | awk 'NR==1{for(i=2;i<=11;i++) printf $i}')` handle the first line of data and binding then from column 2 to 11 since we write 10 byte long data into EEPROM.
And this part of the code could be simplified to something like `awk 'NR==1{printf $1}' "$eeprom"` and it will get the same result. 
I would like to change this part to something like this.
`cat "$eeprom" | od -c ` dump human-readable data from EEPROM as a log for this job.
And simplify read_data to `read_data=$(awk 'NR==1{printf $1}' "$eeprom")`